src/App.js

```diff
@@ -1,13 +1,67 @@
 import React from 'react';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRed } from './api/goods';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+class App extends React.Component {
+  state = {
+    rightGoods: [],
+    selectedFilter: getAll,
+  }
+
+  componentDidUpdate() {
+    this.state.selectedFilter()
+      .then((goodsFromServer) => {
+        this.setState({
+          rightGoods: [...goodsFromServer],
+        });
+      });
+  }
```

Comment:
don't do it in componentDidUpdate
let's create function getGoods, which will accept a callback (filter func) and call and then set to the state goods